cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,278 @@
+
+        old_cert_file.write(certificate)
+        old_cert_file.flush()
+
+        old_key_file.write(private_key)
+        old_key_file.flush()
+
+        # pylint: disable=line-too-long
+        access_token_response = requests.post(
+            token_url,
+            cert=(old_cert_file.name, old_key_file.name),
+            data={
+                "grant_type": "client_credentials",
+                "client_id": client_id,
+            },
+            timeout=30,
+        ).json()
+
+        return access_token_response["access_token"]
+
+
+# TODO(kacpermalachowski): Move it to common package
+def prepare_log_fields() -> Dict[str, Any]:
+    """prepare_log_fields prapares basic log fields"""
+    log_fields: Dict[str, Any] = {}
+    request_is_defined = "request" in globals() or "request" in locals()
+    if request_is_defined and request:
+        trace_header = request.headers.get("X-Cloud-Trace-Context")
+        if trace_header and project_id:
+            trace = trace_header.split("/")
+            log_fields["logging.googleapis.com/trace"] = (
+                f"projects/{project_id}/traces/{trace[0]}"
+            )
+    log_fields["Component"] = "signify-certificate-rotator"
+    log_fields["labels"] = {"io.kyma.component": "signify-certificate-rotator"}
+    return log_fields
+
+
+# TODO(kacpermalachowski): Move it to common package
+def get_pubsub_message():
+    """Parses the Pub/Sub message from the request."""
+    envelope = request.get_json()
+    if not envelope:
+        # pylint: disable=broad-exception-raised
+        raise ValueError("No Pub/Sub message received")
+
+    if not isinstance(envelope, dict) or "message" not in envelope:
+        # pylint: disable=broad-exception-raised
+        raise ValueError("Invalid Pub/Sub message format")
+
+    return envelope["message"]
+
+
+# TODO(kacpermalachowski): Move it to common package
+def prepare_error_response(err: str, log_fields: Dict[str, Any]) -> Response:
+    """Prepares an error response with logging."""
+    _, exc_value, _ = sys.exc_info()
+    stacktrace = repr(traceback.format_exception(exc_value))
+    print(
+        LogEntry(
+            severity="ERROR",
+            message=f"Error: {err}\nStack:\n {stacktrace}",
+            **log_fields,
+        )
+    )
+    resp = make_response()
+    resp.content_type = "application/json"
+    resp.status_code = 500
+    return resp
+
+
+def get_secret(secret_id: str):
+    """Retrieves the latest version of the secret from Secret Manager"""
+    client = secretmanager.SecretManagerServiceClient()
+
+    response = client.access_secret_version(name=f"{secret_id}/versions/latest")
+    secret_value = response.payload.data.decode("UTF-8")
+
+    return json.loads(secret_value)
+
+
+def set_secret(secret_id: str, data: str):
+    """Adds a new version of the secret in Secret Manager."""
+    client = secretmanager.SecretManagerServiceClient()
```

Comment:
Dependency Inversion Principle violation. Please create standalone class with SecretManagerClient with methods get_secret and set_secret.